src/charm.py

```diff
@@ -282,6 +312,16 @@ def _on_upgrade(self, _):
         # force conflict resolution in K8S resources update
         self._on_event(_, force_conflicts=True)
 
+    def _on_update_status(self, event):
+        """Update status actions."""
+        self._on_event(event)
+
+        try:
+            self._refresh_status()
+        except ErrorWithStatus as err:
+            self.model.unit.status = err.status
+            self.logger.error(f"Failed to handle {event} with error: {err}")
```

Comment:
May be remove the error message. There are error logs in `refresh_status` that will provide enough info.